File: Lecture-Passion/src/services/mapping/bookMapping.ts
import { Book } from '@/model/bo/book';

export const mapToBook = (apiData: any): Book => {
  return {
    id: apiData.id,
    title: apiData.title,
    nbPage: apiData.nbPage,
    resume: apiData.resume,
    editionYear: new Date(apiData.editionYear),
    cover: apiData.cover,
    category: apiData.categoryName,
    author: {
      id: apiData.authorId,
      name: apiData.authorName,
    },
    publisher: apiData.publisherName,
  };
};

File: API-ASP-Book/API-ASP-Book/Application/Mappings/BookMapping.cs
public static class BookMapping
{
    public static BookDto ToDto(this BookEntity book)
    {
        return new BookDto
        {
            Id = book.IdBook,
            Title = book.Title,
            AuthorId = book.FkAuthor,
            AuthorName = book.Author.FirstName + " " + book.Author.LastName,
            PublisherId = book.FkPublisher,
            PublisherName = book.Publisher.Name,
            CategoryId = book.FkCategory,
            CategoryName = book.Category.Name,
            EditionYear = (DateTime)book.EditionYear,
            Resume = book.Resume,
            NbPage = book.NbPage,
            Cover = book.Cover
        };
    }
}